opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
I agree with @objectiser and @yurishkuro on that if someone wants ref counting it should be requested explicitly. Without that then this PR isn't really decoupling the activation/deactivation from span lifecycle, it's just providing an opt-out mechanism.

> We will need to document at the API level what it means when no observer is provided (i.e., if it implies no-auto-finish or yes-auto-finish)

Well, shouldn't it be as simple as nothing happens? When I saw the observers in this PR it seemed to me like a general purpose abstraction that -among many other things- can be used for automatically finishing active Spans, but definitely not the main/only problem that can be solved with it. (TBH the first thing that came to my mind is having observers with log statements to debug problems with in-process context propagation). It would be nice if documentation can use automatic finishing as an example of something that could be achieved with Observers, but nothing else.

I understand that such change breaks instrumentation based on 0.30, but whatever version comes next is very likely to be a breaking change so, better get over with it at once.. 0.30.0 has been out for less than two months and probably is not widely spread yet (assumption based on a quick [github search](https://github.com/search?p=1&q=%22opentracing-java%22+%220.30.0%22&type=Code&utf8=%E2%9C%93).